test/gtest/unit_lib_env_var.cpp

```diff
@@ -0,0 +1,247 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2025 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <gtest/gtest.h>
+#include <miopen/errors.hpp>
+#include "../lib_env_var.hpp"
+
+namespace {
+
+MIOPEN_LIB_ENV_VAR(MIOPEN_DEBUG_UNKNOWN_ENVIRONMENT_VARIABLE)
+
+// If these env variables are removed from the library, they need to be replaced with some others.
+MIOPEN_LIB_ENV_VAR(MIOPEN_DEBUG_ENABLE_DEPRECATED_SOLVERS) // some bool variable
+MIOPEN_LIB_ENV_VAR(MIOPEN_DEBUG_TUNING_ITERATIONS_MAX)     // some uint64 variable
+MIOPEN_LIB_ENV_VAR(MIOPEN_DEBUG_FIND_ONLY_SOLVER)          // some string variable
+
+class EnvVarRestorer
+{
+public:
+    EnvVarRestorer(std::string_view name_in) : name(name_in)
+    {
+        prev = miopen::debug::env::GetEnvVariable(name);
+    }
+
+    ~EnvVarRestorer()
+    {
+        if(prev.has_value())
+            miopen::debug::env::UpdateEnvVariable(name, prev.value());
+        else
+            miopen::debug::env::ClearEnvVariable(name);
+    }
+
+private:
+    std::string_view name;
+    std::optional<std::string> prev;
+};
+
+} // namespace
+
+TEST(CPU_UnitTestLibEnvVarGetUnknownVariable_NONE, LibEnvVar)
```

Comment:
Is it possible to use TEST_P and suite instead of plain TEST macro, so we have better control on Smoke/Full runs?

Reply:
Yes, it is possible. Already done.